contributing.md

```diff
@@ -187,6 +191,107 @@ conda create \
         legate-boost
 ```
 
+## Releasing
+
+NOTE: some steps in this section require direct write access to the repo (including its `main` branch).
+
+### Create a stable release
+
+1. Create a pull request updating the `VERSION` file on the `main` branch to the desired version, with no leading `v`
+
+```shell
+echo "24.09.00" > ./VERSION
+```
+
+2. Merge that pull request
+3. Push a git tag like `v24.09.00` ... that tag push will trigger a new release
+
+```shell
+git checkout main
+git pull upstream main
+git tag -a v24.09.00 -m 'v24.09.00'
+git push upstream 'v24.09.00'
+```
+
+4. Update the `VERSION` file again, to the base version for the anticipated next release. Push this directly to `main`, with a commit that includes `[skip ci]` in the message so new packages will not be built from it.
+
+```shell
+git checkout main
+git pull upstream main
+echo "24.12.00" > ./VERSION
+git commit -m "start v24.12 development [skip ci]"
+git push upstream main
+```
+
+5. Tag that commit with a dev version
+
+```shell
+git tag -a v24.12.00dev -m "v24.12.00dev"
+git push upstream v24.12.00dev
+```
+
+From that point forward, all packages produced by CI from the `main` branch will have versions like `v24.12.00.dev{n}`,
+where `{n}` is "number of new commits since the one tagged `v24.12.00.dev`".
+
+### Hotfixes
+
+Imagine that `v24.09.00` has been published, and at some later point a critical bug is found, which you want to package and release as `v24.09.01`.
+
+Do the following.
+
+1. Create a release branch, cut from the tag corresponding to the release you want to fix.
+
+```shell
+# get all the tags locally
+git checkout main
+git pull upstream main --tags
+
+# create the new branch
+git checkout v24.09.00
+git checkout -b release/24.09
+echo 'v24.09.01' > ./VERSION
+git commit -m 'start v24.09.01 [skip ci]'
+git push upstream release/24.09
+
+# tag the first commit on the new branch as the beginning of the 24.09.01 series
+git tag -a v24.09.01dev -m 'v24.09.01dev'
+git push upstream v24.09.01dev
+```
+
+2. Open pull requests targeting that branch and merge them into that branch.
+3. When you feel the branch is ready to release, push a new tag.
+
+```shell
+git checkout release/v24.09
+git pull upstream release/v24.09 --tags
+git tag -a v24.09.01 -m 'v24.09.01'
+git push upstream v24.09.01
+```
+
+With that hotfix release complete, merge the fixes into `main`.
```

Comment:
In RAPIDS, no developer ever targets a `main` branch directly. RAPIDS projects one long-lived branch per release (e.g. `branch-24.10`, `branch-24.12`).

Changes on those release branches are not merged to `main` until the entire branch is released.

A hotfix to the 24.10 release after work on the 24.12 release has already begun means new pull requests targeting `branch-24.10`. Those are then merged to `main`, a new `v24.10.01` tag is cut (pointing at a commit on `main`), and hotfix packages are released.

It's rare (I think) that we apply hotfixes more than 1 major release back (e.g., releasing a `cudf==24.10.02` when there is already a stable `cudf==24.12.00`). But if we did, then those tags could point to commit IDs that exist on `branch-24.10` but not on `main`.

Here, I'm proposing using a different, every-PR-targets-`main` approach for the following reasons:

* to avoid the pain of having long-lived release branches (until we find we need one, for hotfixes)
* to test out workflows to serve as an example for maybe changing the RAPIDS flow in the future: https://github.com/rapidsai/build-planning/issues/63
* for consistency with how `cunumeric` and `legate` operate

Reply:
Great.  I half thought I heard rumblings for rapids to switch to this branching model, but maybe I am hallucinating :).